design/application-abi.md

```diff
@@ -15,10 +15,17 @@ Current Unstable ABI
 There are two kinds of modules:
 
  - A *command* exports a function named `_start`, with no arguments and no return
-   values. Environments shall call this function once, after instantiating the
-   module and all of its dependencies. After this function exits, the instance
-   is considered terminated and no further use of any of its exports should be
-   permitted.
+   values.
+
+   `_start` is the default export which is called when the user doesn't select a
+   specific function to call. Commands may also export additional functions,
+   (similar to "multi-call" executables), which may be explicitly selected by the
+   user to run instead.
+
+   Functions exported from a command are available to be called without a
+   pre-existing instance. When they are called, the module is instantiated and used
```

Comment:
This doesn't make a lot of sense in terms of the current JS embedding where an instance is required to even get hold of the export.   Even the current node wasi implementation requires in instance before its exports can be used.    Perhaps we could re-word to keep the intent but allow for such implementations?

How about something like `Functions exported from a command are required to run in a fresh instance each time they are called, and when the call returns, the instance is considered terminated and should not be accessed.`

Reply:
That's a good point; I've reworded this section to phrase it in terms of what instances may assume rather than in terms of how instantiation actually works.